src/prefect/new_flow_engine.py

```diff
@@ -163,21 +211,21 @@ async def create_flow_run(self, client: PrefectClient) -> FlowRun:
         flow_run_ctx = FlowRunContext.get()
 
         parent_task_run = None
+
         # this is a subflow run
         if flow_run_ctx:
+            self.is_subflow = True
+
+            # get the parent task run
             parent_task_run = await self.create_subflow_task_run(
                 client=client, context=flow_run_ctx
             )
-            # If the parent task run already completed, return the last flow run
-            # associated with the parent task run. This prevents rerunning a completed
-            # flow run when the parent task run is rerun.
-            most_recent_flow_run = (
-                await self.get_most_recent_flow_run_for_parent_task_run(
-                    client=client, parent_task_run=parent_task_run
-                )
```

Comment:
`get_most_recent_flow_run_for_parent_task_run` looks like it isn't used anywhere else can can be removed

Reply:
Good call